### PR TITLE
Upload list activity: new sort / switch button

### LINKED ISSUE #6202: Sort-by and grid/listview not effective in Local synchronization
Hi,

In this view (Setup a custom folder) :
 
Capture d’écran 2020-06-02 à 17 20 46

When you choose a local folder, there are buttons to sort-by/change display but they don't work : 

bug

Based on master.
Due to : #5207

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix #6202

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

**Comment by Shagequi** [comment]
Seems good to me 👍 
Thank you for this PR, I was going to work on it this morning ^^

**Comment by tobiasKaminsky** [comment]
> Thank you for this PR, I was going to work on it this morning ^^

:-D
Once all PRs regarding those buttons are merged, I will eliminate all code references to it (if any).

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 1]
```diff
@@ -796,26 +784,20 @@ public void onPrepareOptionsMenu(@NonNull Menu menu) {
                 break;
 
             case REMOVE_SEARCH:
-                menu.removeItem(R.id.action_search);
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java:799]: Why you remove search ?

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 2]
```diff
@@ -1287,7 +1269,7 @@ private void updateLayout() {
             switchToListView();
         }
 
-        setSortButton(preferences.getSortOrderByFolder(mFile));
+        DisplayUtils.setSortButton(mSortButton, preferences.getSortOrderByFolder(mFile), getContext());
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java:1272]: I prefere use "requireContext()" :)
requireContext() will throw an exception if fragment isn’t attached to its host

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextFileFragment.java [hunk 3]
```diff
@@ -287,12 +287,10 @@ public void onPrepareOptionsMenu(@NotNull Menu menu) {
...
         );
 
-        Boolean dualPane = getResources().getBoolean(R.bool.large_land_layout);
+        boolean dualPane = getResources().getBoolean(R.bool.large_land_layout);
 
         if (!dualPane) {
-            FileMenuFilter.hideMenuItems(menu.findItem(R.id.action_switch_view),
-                                         menu.findItem(R.id.action_sort)
-            );
+            FileMenuFilter.hideMenuItems(menu.findItem(R.id.action_sort));
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/preview/PreviewTextFileFragment.java:293]: "action_sort" is not removed?

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextStringFragment.java [hunk 4]
```diff
@@ -145,7 +145,6 @@ public void onPrepareOptionsMenu(@NonNull Menu menu) {
         super.onPrepareOptionsMenu(menu);
 
         menu.findItem(R.id.action_sort).setVisible(false);
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/preview/PreviewTextStringFragment.java:147]: "action_sort" is not removed?

File: src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java [hunk 5]
```diff
@@ -170,15 +172,32 @@ public void onCreate(Bundle savedInstanceState) {
...
+        uploadButton.setTextColor(ThemeUtils.fontColor(this, false));
+        uploadButton.setOnClickListener(this);
+
+        sortButton = findViewById(R.id.sort_button);
+        FileSortOrder fileSortOrder = preferences.getSortOrderByType(FileSortOrder.Type.uploadFilesView);
+        DisplayUtils.setSortButton(sortButton, fileSortOrder, this);
+        sortButton.setOnClickListener(l -> openSortingOrderDialogFragment(getSupportFragmentManager(), fileSortOrder));
+
+        MaterialButton switchButton = findViewById(R.id.switch_grid_view_button);
+        switchButton.setOnClickListener(l -> {
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java:191]: We can not extract this in DisplayUtils? OCFileListFragment use setGridSwitchButton :)

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java:191]: Check if is a grid view is different, so we cannot use it 1:1.

File: src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java:799]: No idea :see_no_evil: 
Added back :+1:

File: src/main/java/com/owncloud/android/ui/preview/PreviewTextFileFragment.java (hunk 3 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/ui/preview/PreviewTextFileFragment.java:293]: Ops.
After all changes I will go over whole project and remove all action_sort action_switch…
Thanks for finding

File: src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java (hunk 5 above)
Comment (JorisBodin) [reply src/main/java/com/owncloud/android/ui/activity/UploadFilesActivity.java:191]: For setIcon no?

**Pull request merged**